src/main/scala/io/github/pauljamescleary/petstore/infrastructure/authentication/UserBackingStore.scala

```diff
@@ -0,0 +1,23 @@
+package io.github.pauljamescleary.petstore
+package infrastructure.authentication
+
+import cats.data.OptionT
+import tsec.authentication.BackingStore
+import domain.users.{User, UserRepositoryAlgebra}
+
+abstract class UserBackingStore[F[_]] extends BackingStore[F, Long, User]
+
+object UserBackingStore {
+
+  /**
+    * A transformation between the UserRepositoryAlgebra in our domain and a BackingStore
+    * for users which is required by tsec at our endpoint.
+    */
+  def trans[F[_]](A: UserRepositoryAlgebra[F]): UserBackingStore[F] = new UserBackingStore[F] {
```

Comment:
I think this could be named a little better, but can't come up with anything right now. I'm using it as `UserBackingStore.trans(userAlgebra)`, so maybe that's good.

I think this addresses the code-reuse vs SoC dilemma ok, let me know what you think.